### PR TITLE
Better handling of API error messages

### PR DESCRIPTION
**Pull request by hmarr**
- Detect errors under the `errors` key, as well as under `error`
- Handle field errors that come as a hash, as well as arrays of error messages

Review please @petehamilton.

### CODE REVIEW

**Comment by petehamilton** [comment]
@hmarr Does this still need review?

**Comment by hmarr** [comment]
Yes please!

File: lib/gocardless/errors.rb [hunk 1]
```diff
@@ -10,15 +10,34 @@ class ApiError < Error
     def initialize(response)
       @response = response
       @code = response.status
-      body = MultiJson.decode(response.body) rescue nil
-      if body.is_a? Hash
-        @description = body["error"] || response.body.strip || "Unknown error"
+
+      begin
+        parsed_response = MultiJson.decode(response.body)
+        errors = parsed_response["error"] || parsed_response["errors"]
```
Comment (petehamilton) [inline lib/gocardless/errors.rb]: :sob:

**Comment by petehamilton** [comment]
:+1:

**Pull request merged**